src/components/HbLoadingSpinner.tsx

```diff
@@ -0,0 +1,90 @@
+import random from "lodash/random";
+import React, { ReactNode, useContext, useMemo } from "react";
+import { Css } from "src/Css";
+import SpinnerGifBase64 from "./HbLoadingSpinner.base64";
+
+interface HbLoadingSpinnerProps {
+  /** Reverts loading text to `Loading...` if true. May override global noQuips by passing in noQuips={false}. */
+  noQuips?: boolean;
+  /** Contextually relevant quips, like "Where did I put that invoice??" on some invoice page. Will override global noQuips if non-empty. */
+  extraQuips?: ReadonlyArray<string>;
+  extraQuipsOnly?: boolean;
+  /** No text at all */
+  iconOnly?: boolean;
+}
+
+export function HbLoadingSpinner({ noQuips, extraQuips = [], extraQuipsOnly, iconOnly }: HbLoadingSpinnerProps) {
+  const ctx = useContext(HbLoadingSpinnerContext);
+
+  const quip = useMemo(() => {
+    // If quips are off globally but custom quips were provided, then override global
+    const forceQuips = extraQuips.length !== 0 || noQuips === false;
+    if ((ctx.noQuips && !forceQuips) || noQuips) {
+      return "Loading...";
+    }
+
+    const allQuips = extraQuipsOnly && extraQuips.length !== 0 ? extraQuips : [...ctx.quips, ...extraQuips];
+    return allQuips[random(allQuips.length - 1)];
+  }, [ctx.noQuips, ctx.quips, extraQuips, extraQuipsOnly, noQuips]);
+
+  return (
+    <div data-testid="hbLoadingSpinner" css={Css.df.fdc.jcc.aic.$}>
+      <img
+        src={SpinnerGifBase64}
+        data-chromatic="ignore" // Chromatic timing would snapshot this on different frames of the 24-frame gif, so ignore it
```

Comment:
Huh, does this really work?

Reply:
According to their docs yeah. According to real-world experience, unknown.

Reply:
With both of these elements being ignored in chromatic, maybe we can just ignore the stories all together?
```
export const StoryName = setStory(...);
StoryName.parameters = {
  // disables Chromatic's snapshotting on a story level
  chromatic: { disableSnapshot: true },
};
```

Reply:
Or better yet:
```
export default {
  component: MyComponent,
  // Disables snapshotting for the component
  parameters: {
    chromatic: { disableSnapshot: true },
  },
  title: 'MyComponent',
};
```

Reply:
The other issue to consider is Consumers. i.e. BluePrint or Underwriting snapshots shouldn't be triggering because someone decided to use this somewhere.

I can control Beam's stories easily enough but I don't want BP/UW/HOP/etc to have to jump through hoops as well. So I still need to `data-chromatic="ignore"`

Reply:
Smart!